sound/soc/sof/loader.c

```diff
@@ -546,10 +546,10 @@ int snd_sof_run_firmware(struct snd_sof_dev *sdev)
 				 msecs_to_jiffies(sdev->boot_timeout));
 	if (ret == 0) {
 		dev_err(sdev->dev, "error: firmware boot failure\n");
-		/* after this point FW_READY msg should be ignored */
-		sdev->boot_complete = true;
 		snd_sof_dsp_dbg_dump(sdev, SOF_DBG_REGS | SOF_DBG_MBOX |
 			SOF_DBG_TEXT | SOF_DBG_PCI);
+		/* after this point FW_READY msg should be ignored */
+		sdev->boot_complete = true;
```

Comment:
@plbossart just wondering why do we need to set this to true at all in case of FW boot failure?

Reply:
@kv2019i Can you chime in, this change was yours initially I believe and IIRC it was intentional. I don't recall the reason...

Reply:
@plbossart i looked up the commit that added the change a69270d8bfeb3. I think @kv2019i hit this race condition with the QEMU DSP. But nevertheless, this change looks fine to me.

Reply:
@plbossart @ranj063 Ack on that.  Without that you can have the case where we consider FW loading to have failed to a timeout and have freed, or are freeing related resources -- but then we get the late FW_READY and this will make the driver go crazy. Thus the fix.